### PR TITLE
GO-3031: allow ObjectCreateFromUrl from url to file

### PR DESCRIPTION
**Pull request by AnastasiaShemyakinskaya**
https://linear.app/anytype/issue/GO-3031/allow-objectcreatefromurl-from-url-to-file

### CODE REVIEW

File: util/linkpreview/linkpreview.go [hunk 1]
```diff
@@ -40,7 +41,7 @@ const (
 var log = logging.Logger("link-preview")
 
 type LinkPreview interface {
-	Fetch(ctx context.Context, url string) (model.LinkPreview, []byte, error)
+	Fetch(ctx context.Context, url string) (model.LinkPreview, []byte, bool, error)
```
Comment (KirillSto) [inline util/linkpreview/linkpreview.go:44]: We can add comment/names for return values here as well

File: util/linkpreview/linkpreview_test.go [hunk 2]
```diff
@@ -23,7 +23,7 @@ func TestLinkPreview_Fetch(t *testing.T) {
 		lp := New()
 		lp.Init(nil)
 
-		info, _, err := lp.Fetch(ctx, ts.URL)
+		info, _, _, err := lp.Fetch(ctx, ts.URL)
```
Comment (KirillSto) [inline util/linkpreview/linkpreview_test.go:26]: Should not we check isFile value here?

File: util/linkpreview/linkpreview.go [hunk 3]
```diff
@@ -170,6 +172,17 @@ func (l *linkPreview) makeNonHtml(fetchUrl string, resp *http.Response) (i model
...
 }
 
+func checkFileType(url string, resp *http.Response, contentType string) bool {
+	ext := filepath.Ext(url)
+	mimeType := mime.TypeByExtension(ext)
+	return isContentFile(resp, contentType, mimeType)
+}
+
+func isContentFile(resp *http.Response, contentType, mimeType string) bool {
+	return contentType != "" || strings.Contains(resp.Header.Get("Content-Disposition"), "filename") ||
```
[the remark this answers is not in the document]
Comment (AnastasiaShemyakinskaya) [reply util/linkpreview/linkpreview.go:182]: It's checked in upper function

File: util/linkpreview/cache.go [hunk 4]
```diff
@@ -32,11 +32,11 @@ func (c *cache) Name() string {
 	return CName
 }
 
-func (c *cache) Fetch(ctx context.Context, url string) (lp model.LinkPreview, body []byte, err error) {
+func (c *cache) Fetch(ctx context.Context, url string) (lp model.LinkPreview, body []byte, b bool, err error) {
```
Comment (KirillSto) [inline util/linkpreview/cache.go:35]: **b** should be also renamed to **isFile**, isn't it?

**Review by KirillSto — APPROVED** [review]

File: util/linkpreview/cache.go (hunk 4 above)
Comment (AnastasiaShemyakinskaya) [reply util/linkpreview/cache.go]: Fixed

**Pull request merged**